Initial proposal for visual redesign (#16)
osu-tournament-rating/otr-wiki: api/endpoints/matches/en.md

# Matches

### Submit Matches for Verification

( **user** ) ( **submit** )\
Submit a tournament and associated matches to be verified for rating calculation

Request\
( **POST** ) `/matches/batch`

Query Parameters\
`verified` boolean *optional*\
Setting `verified` to `true` will assert that the `submitterId` has the ( **verifier** ) scope

Body Parameters\
See [TournamentWebSubmission](/api/objects/en.md#tournamentwebsubmission)

Example Request
```js
const requestBody = JSON.stringify({
  "name": "5 Digit World Cup",
  "abbreviation": "5WC",
  "forumPost": "https://osu.ppy.sh/community/forums/topics/1874523",
  "rankRangeLowerBound": 10000,
  "mode": 0,

>   "teamSize": 8,

Reviewer: ```suggestion
  "teamSize": 4,
```
🤓

# Matches

### Submit Matches for Verification
( **PUBLIC** )\
Submit a tournament and it's matches to be verified for rating calculation

Request\
( **POST** ) `/matches/batch`

Query Parameters\
`verified` boolean *optional*\

> Setting this to true will verify that the `submitterid` is an ( **ADMIN** ) user. Will result in a 401 if not

Reviewer 2: Language should be consistent and more direct with direct references to variables and adding `code marks`. `camelCase` should be used when demonstrating most API query functions and responses. I also don't know how we should be writing user scopes here, but I have corrected the scope from admin to verifier. Scopes should always be identical in reference to [this page](https://github.com/osu-tournament-rating/otr-wiki/blob/master/api/scopes/en.md), including casing. We should definitely add a separate page specifically describing all of these documentation quirks so we can remain consistent when documenting.

```suggestion
Setting `verified` to `true` will assert that the `submitterId` has the ( **verifier** ) scope.
```

Instead of saying `Will result in a 401`, we should separately document response codes in a more concrete fashion.

This comment applies to all API documentation in this PR.

Author: > Language should be consistent and more direct with direct references to variables and adding `code marks`.

Agree

> `camelCase` should be used when demonstrating most API query functions and responses.

Also agree, I wasn't initially sure how certain fields translated from the api code to fields in the responses. In the code all variables for query parameters and body parameters are `camelCase`, but all fields in the DTOs are `PascalCase`. I understand they all get converted to `camelCase` regardless?

> I also don't know how we should be writing user scopes here, but I have corrected the scope from admin to verifier. Scopes should always be identical in reference to [this page](https://github.com/osu-tournament-rating/otr-wiki/blob/master/api/scopes/en.md), including casing.

Yep this was all written out before you did any documentation on scopes and I was purposefully trying to keep it vague awaiting the OAuth migration

> We should definitely add a separate page specifically describing all of these documentation quirks so we can remain consistent when documenting.

I'll open up an issue about wiki meta docs then, don't know if it would belong in this pull

Body Parameters\
See [TournamentWebSubmission](/api/objects/en.md#tournamentwebsubmission)

Response Format\

> Http Response ( 200 | 400 | 401 )

Reviewer 2: Write `HTTP` instead
```suggestion
HTTP Response ( 200 | 400 | 401 )
```

Example Request
```js
const requestBody = JSON.stringify({
  "name": "5 Digit World Cup",
  "abbreviation": "5WC",
  "forumpost": "https://osu.ppy.sh/community/forums/topics/1874523",
  "rankrangelowerbound": 10000,
  "mode": 0,
  "teamsize": 8,
  "submitterid": 18068913,
  "ids": [18068913, 18068914, 18068915, ...],

> });

Reviewer 2: Everything here should be `camelCase`

> fetch("https://api.otr.xyz/matches/batch", {

Reviewer 2: Endpoint incorrect. Consider defining a root once so we don't have to type this url over and over.

```suggestion
fetch("https://api.otr.stagec.xyz/matches/batch", {
```

Author: Will definitely define a root. I actually just took a stab and guessed at the domain so I guess I was pretty close lol

  method: "POST",
  body: requestBody,
});
```

---

### Refresh Automation Checks

( **ADMIN** )\
Marks all matches as needing automation checks

Request\
( **POST** ) `/matches/refresh/automations`

Response Format\

> Http Response ( 200 | 401 )

Reviewer 2: Technically this does not return a `401`. `429` is returned by the API middleware if the user has insufficient permissions to execute this command.

I think it'd be best to document middleware results (such as `429`) once somewhere important, then omit them at the endpoint level. 

Adjust anywhere else an implied 401 is used without explicit documentation / where the function does not ever return `BadRequest()`.

Author: Good idea, I'll open a separate issue for middleware responses then.

---

### Get All Verified Match Ids

( **ADMIN** )\
Returns a list of all verified match ids

Request\
( **GET** ) `/matches/ids`

Response Format
| Field | Type |
| ----- | ---- |

> | { ? } | integer[] |

Reviewer 2: When an endpoint returns any generic collection, the field is unnamed. I imagine this is a fine way to indicate this, but just so you know.

Author: 👍

> ### Convert a List of Match Ids to Match Objects

Reviewer 2: Capitalization is kinda all over the place. I'd capitalize the first word & class names only. Check all the grammar in similar headings.

Author: All of the headings are in [title case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case), save for  words like `Ids` which would count as proper in this context I believe

Reviewer 2: Ah ok, as long as there is a consistent scheme, that's fine.
